### PR TITLE
very slight optimization that could make a difference

### PR DESCRIPTION
**Pull request by irusei**
I really don't know much about gmod lua but this should run better and maybe give more fps idk

I also have not tested this a lot however it did work for me

### CODE REVIEW

File: lua/autorun/client/TrueCulling.lua [hunk 1]
```diff
@@ -3,53 +3,31 @@ jit.on()
...
-	allEnts = ents.GetAll()
 	if engine.TickCount()%4 == 0 then
+		allEnts = ents.GetAll()
 		for a, ents_ in pairs(allEnts) do
 			if engine.TickCount()%10 == 0 then
 				ents_:SetNoDraw(true)
-			end
-			if engine.TickCount()%10 == 0 then
-				if ents_:GetNoDraw(true) then
+				if ents_:GetNoDraw() then
```
Comment (shiirudev) [inline lua/autorun/client/TrueCulling.lua:11]: I do not know if this if is necessary but let's keep it in I guess

File: lua/autorun/client/TrueCulling.lua [hunk 2]
```diff
@@ -8,6 +8,9 @@ function main()
 		for a, ents_ in pairs(allEnts) do
 			if engine.TickCount()%10 == 0 then
 				ents_:SetNoDraw(true)
+				if ents_:GetClass() == "10C_BaseFlex" then
+					continue
```
Comment (shiirudev) [inline lua/autorun/client/TrueCulling.lua:12]: should work if this does what it does in every single other programming language

Comment (insertname1180) [reply lua/autorun/client/TrueCulling.lua:12]: @shiirudev Hang tight, will get to you in a moment. Just going to review your code :+1:

**Comment by insertname1180** [comment]
@shiirudev Hang tight, just reviewed your code... so LuaJIT in Garry's Mod is interesting due to it being single threaded. So the scuffed reason why there is so many ```if engine.TickCount()%10 == 0 then``` is because Lua waits that long until it executes in Garry's Mod's context. If it was multi-threaded then they would all execute at the same time. The interpreter goes down line by line in the code so it'll wait 10 ticks until the code can execute each time ```if engine.TickCount()%10 == 0 then``` is used, because it is single threaded. I don't know why it's single-threaded... it just is. So the main point is the code doesn't execute at the same time but instead in segments so the frame drops are less worse. For some reason as well, ```allEnts = ents.GetAll()``` getting called every tick is actually better... I'm curious myself as to why.

**Comment by irusei** [comment]
weird, I have personally not experienced any frame drops with this and even have gotten like 5% more fps

File: lua/autorun/client/TrueCulling.lua [hunk 3]
```diff
@@ -8,7 +8,7 @@ function main()
 		for a, ents_ in pairs(allEnts) do
 			if engine.TickCount()%10 == 0 then
 				ents_:SetNoDraw(true)
-				if ents_:GetClass() == "10C_BaseFlex" then
+				if ents_:GetClass() == "10C_BaseFlex" or ents_:GetClass() == "class C_BaseFlex" then
```
Comment (shiirudev) [inline lua/autorun/client/TrueCulling.lua:11]: could be good to also add gmod_hands here

**Comment by insertname1180** [comment]
I'm curious if your % lows are better or worse, might be worth looking into?

Edit: I think your pull request is great! Going to test and see if there is any performance issues... it's a bit odd how there's a difference between machines. I'll do some more testing... out of curiousity what's your OS? I'm running Linux, I'll try proton to see if there is any difference in performance. I very much appreciate your work!

**Comment by insertname1180** [comment]
> @shiirudev Hang tight, just reviewed your code... so LuaJIT in Garry's Mod is interesting due to it being single threaded. So the scuffed reason why there is so many `if engine.TickCount()%10 == 0 then` is because Lua waits that long until it executes in Garry's Mod's context. If it was multi-threaded then they would all execute at the same time. The interpreter goes down line by line in the code so it'll wait 10 ticks until the code can execute each time `if engine.TickCount()%10 == 0 then` is used, because it is single threaded. I don't know why it's single-threaded... it just is. So the main point is the code doesn't execute at the same time but instead in segments so the frame drops are less worse. For some reason as well, `allEnts = ents.GetAll()` getting called every tick is actually better... I'm curious myself as to why.

This is actually not relevant anymore. I'm doing a huge rewrite of the code.

**Pull request closed without merging**